renpy/script.py

```diff
@@ -1117,7 +1136,10 @@ def lookup(self, label):
             rv = self.namemap.get(label, None)
 
         if rv is None:
-            raise ScriptError("could not find label '%s'." % str(original))
+            if isinstance(original, str):
+                raise LabelNotFound(original)
+            else:
+                raise ScriptError(f"could not find label '{original}'.")
```

Comment:
Feels pretty icky to have the exception type governed by whether we're looking up a string vs anything else. Perhaps this check should reside in `LabelNotFound.get_suggestion`?